packages/calypso-e2e/src/lib/pages/revisions-page.ts

```diff
@@ -39,30 +38,28 @@ export class RevisionsPage {
 		// slider width.
 		const clickPositionX = ( sliderWidth * ( index - 1 ) ) / ( revisionCount - 1 );
 
-		await Promise.all( [
-			// We need to rely on waiting for mutations here because neither
-			// checking the ".loading" status nor the auto-waiting are enough to
-			// indicate that the diff has been loaded and is ready to be
-			// interacted with. This might be related to the fact that the diff
-			// is rendered entirely from a fetched HTML string.
-			ElementHelper.waitForMutations( this.page, '.revisions-diff-frame' ),
-			slider.click( { position: { x: clickPositionX, y: 1 } } ),
-		] );
+		await this.page.waitForLoadState( 'networkidle' );
+		await slider.click( { position: { x: clickPositionX, y: 1 } } );
```

Comment:
This is my workaround to the complicated `waitForMutation` call - wait until the network activity has ceased.

`networkidle` has its own set of issues such as when pages inadvertently take longer than expected to load. On the other hand, it's simple to understand and follow along. Any timeouts we can address by adding an appropriate timeout.

Reply:
**Question/Comment (blocking)**: could you explain this change a little bit more? Was the `waitForMutation` call being flaky?

I could be misunderstanding, but I'm not sure this is a comparable change, and I'm not sure this will get us what we want over time. 🤔 

Previously, the `Promise.all` check, combined with the `waitForMutations` call, made sure that we knew the revisions view had finished changing _after_ we used the slider.

Now, we're just waiting for networkidle here _before_ we use the slider. So interacting with the revision panel right after this is still prone to race conditions!

Also, I'm not sure checking for `networkidle` this long after a document load is going to do anything. I'm pretty sure that Playwright looks back to see if that event ever fired for this URL/document, so this may just immediately carry on?

Reply:
My initial understanding was that `waitForMutation` is checking the stability of the page _before_ the revision was loaded. 

With the corrected spec but without removing the `waitForMutation`, the following is seen because it is leaving behind open handles for Jest to clean up.

```
Ran all test suites matching /test\/e2e\/specs\/editor\/editor__revisions.ts/i.
Jest did not exit one second after the test run has completed.

This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.
```

Perhaps I can pose a few questions that will help lead to a decision:
- is there a need to "wait" until the revision is ready to interact, as long as the "load revision" button is ready to interact?
- instead of the complex `waitForMutation`, could we stand in a wait for string we know is part of the revision?

Reply:
> With the corrected spec but without removing the waitForMutation, the following is seen because it is leaving behind open handles for Jest to clean up.

Hmm, weird... 🤔 

Here's what my local testing found:
- Running the test without the `waitForMutations` (as in the current change) actually failed! It looks like it hit the race condition the `waitForMutations` was trying to help with -- the revision content didn't change fast enough, so the validations of that new content ended up failing.
- Running with the `waitForMutations` passed and didn't leave any open handles. 

I wonder if maybe I'm missing something else from your local set-up? Here was my command:
`TEST_ON_ATOMIC=true yarn jest specs/editor/editor__revisions.ts`

> is there a need to "wait" until the revision is ready to interact, as long as the "load revision" button is ready to interact?

"Ready to interact" may be confusing wording here... I think the idea was that we don't want to do validations of the _content of the revisions panel_ until we were sure the new revision content loaded.

> Instead of the complex waitForMutation, could we stand in a wait for string we know is part of the revision?

I think that's a totally valid approach! 👍  I personally don't mind the `waitForMutation` function, and I like that it "future-proofs" a little bit, but we can totally design around this by focusing on specific text content and playwright-waiting for that (instead of getting the `innerText` and doing string assertions against that).

I'm fine either way! 😁  I just think the current approach in the PR might introduce some flakiness.

Reply:
@dpasque I'll revisit or eliminate the `waitForMutation` changes. It was on the table only because Jest reliably (on my machine) threw the open handle warning, but it wasn't strictly necessary to fix the `Editor: Revision` spec.